### ISSUE TITLE
tpm2_getrandom: add single-bytes option

### ISSUE THREAD

**Issue by joholl** [tpm2-software/tpm2-tools#1611]
Prior to 460afc31 `tpm2_getrandom` printed something like this:
```
0x01 0x23 0x45 0x67
```

Now, `tpm2_getrandom` prints either raw bytes or a hex stream (`--hex`):
```
01234567
```

While I very much prefer the hex stream, we should add an option `--single-bytes` (or similar) to provide the output format previously used. This way the user can switch back to the previous output format if needed. In the end, we should not break more than necessary and provide compatibility solutions where possible ;).

**Comment by williamcroberts** [comment]
@joholl we keep backwards compatibility where it makes sense:
1. keeping an old interface requires very little overhead in code quantity or maintenance
2. their were users that really needed it

Considering that every other getrandom routine just outputs it as a byte array, I don't know how useful that old interface is, and if someone was using ossl and they needed that interface they would have already done something like:
```bash
echo "00112233" | sed 's/.\{2\}/0x& /g'
```
Are you actually irreparably broken by that?

The reason we are making such sweeping changes on 4.0 is to kill off the legacy of bad interfaces and choices made in part by myself and predecessors. Moving forward from 4.0 will be a stable interface.

**Comment by joholl** [comment]
I see your point there. Version 4.0 is the opportunity to remove old interfaces. While I would argue that having the old interface does not hurt, it's not a dealbreaker when it's gone.

**Comment by williamcroberts** [comment]
> I see your point there. Version 4.0 is the opportunity to remove old interfaces. While I would argue that having the old interface does not hurt, it's not a dealbreaker when it's gone.

It does hurt in the sense that we would have to keep it and maintain it going forward. If no one was really banking on it, or there is a trivial work around i'd keep it. If we needed backwards compat, we'd have to keep the old behavior and opt into the new string interfaces, so adding an option wouldn't really help. But I appreciate the bug report, at the very least it shows people kicking the tires. I need to know what users problems are, so hopefully we can fix them in 4.0 or make sure we don't do something terribly stupid.
